ttnn/cpp/ttnn/operations/conv/conv_transpose2d/conv_transpose2d.cpp

```diff
@@ -0,0 +1,350 @@
+        ttnn::Shape(std::array<uint32_t, 4>{1, 1, batch_size * output_height * output_width, tt::round_up(out_channels, 32)}),
+        parallel_config,
+        round_up_size);
+        auto opt_conv_op_parallel_config = conv2d::determine_conv_op_parallel_config_from_conv_output_mem_config(
+            conv_out_memory_config,
+            conv2d::get_num_cores_nhw_from_parallel_config(parallel_config),
+            conv2d::get_num_cores_channels_from_parallel_config(parallel_config)
+        );
+        auto opt_conv_op_block_config = conv2d::determine_per_core_conv_block_config(
+            parallel_config,
+            opt_conv_op_parallel_config,
+            tt::round_up(in_channels, conv_config.input_channels_alignment),
+            conv_config.act_block_h_override,
+            conv_config.act_block_w_div,
+            kernel_size[0],
+            kernel_size[1],
+            conv_config.fp32_dest_acc_enabled,
+            conv_config.enable_split_reader);
+
+        //TODO: Flip the Weights
+        bool weight_is_on_device = ttnn::is_tensor_on_device_or_multidevice(weight_tensor);
+        ttnn::Tensor weight_tensor_on_device = weight_tensor;
+        std::optional<ttnn::Tensor> bias_tensor_on_device = bias_tensor;
+        if (!weight_is_on_device) {
+
+            // prepare weights in desired layout and move to device
+            tie(weight_tensor_on_device, bias_tensor_on_device) = conv2d::prepare_conv_weights_biases_and_move_to_device(
+                transform_weights_for_conv_transpose2d(weight_tensor),
+                bias_tensor,
+                conv_config.input_channels_alignment,
+                conv_config.weights_dtype,
+                opt_conv_op_block_config.act_block_w_ntiles,
+                opt_conv_op_block_config.out_subblock_w_ntiles,
+                parallel_config,
+                device,
+                groups,
+                opt_conv_op_block_config.act_block_h_ntiles,
+                input_width);
+        }
+        // call conv micro op
+        auto conv_output = optimized_conv_new(
+            halo_output,
+            weight_tensor_on_device,
+            bias_tensor_on_device,
+            sliding_window_config,
+            out_channels,
+            groups,
+            conv_config.output_layout == Layout::ROW_MAJOR,
+            conv_config.activation == "relu",
+            conv_config.math_fidelity,
+            opt_conv_op_parallel_config,
+            opt_conv_op_block_config,
+            conv_out_memory_config,
+            conv_config.dtype,
+            {batch_size, input_height, input_width, in_channels},
+            conv_config.input_channels_alignment == 16,
+            compute_kernel_config,
+            conv_config.enable_act_double_buffer,
+            conv_config.enable_split_reader,
+            conv_config.enable_subblock_padding);
+        ttnn::operations::core::deallocate(halo_output);
```

Comment:
What's the point of this explicit deallocation if the scope ends in the next line? 
Should be totally fine for halo Tensor to get deallocated in its destructor.

Reply:
Removed.